test/cmd/discovery.sh

```diff
@@ -407,3 +407,43 @@ __EOF__
   set +o nounset
   set +o errexit
 }
+
+run_explain_crd_with_additional_properties_tests() {
+  set -o nounset
+  set -o errexit
+
+  create_and_use_new_namespace
+  kube::log::status "Testing explain with custom CRD that uses additionalProperties as non boolean field"
+
+  output_message=$(kubectl apply -f hack/testdata/CRD/example-crd-with-additionalfields.yaml)
```

Comment:
Could you please move this CRD `hack/testdata/CRD/example-crd-with-additionalfields.yaml` in here by applying it with `EOF` (and also simplify CRD by discarding unnecessary fields). It seems very specific CRD and no need to store it in there.

Reply:
Done